src/components/notice/TopTab.tsx

```diff
@@ -1,49 +1,118 @@
+  const [category, setCategory] = useState("ALL");
+  
+  const [fetching, setFetching] = useState(false); //데이터 받아오는 중인지 표시
+  const [page, setPage] = useState(0);
+  const [search, setSearch] = useState("");
+
+  const fetchNoticeData = async () => {
+
+    try{
+      if(page ===0){
+        setNoticeData([]);
+      }
+      const noticeData = await getNoticeAPIwithCategory(search,category,page);
+      if(noticeData){
+        setNoticeData((prev)=>[...prev,...noticeData.data.content]);
+      }else{
+        console.error('공지사항 데이터를 불러오는 중 오류 발생:', noticeData.message);
+      } 
+      
+    }catch(error){
+      console.error('공지사항 데이터를 불러오는 중 오류 발생:', error);
+    }
+  };
+
+  const fetchFunSystemData = async () => {
+
+    try{
+      if(page ===0){
+        setFunSystemData([]);
+      }
+      const funSystemData = await getFunSystemAPIwithCategory(search,category,page);
+      if(funSystemData){
+        setFunSystemData((prev)=>[...prev,...funSystemData.data.content]);
+      }else{
+        console.error('펀시스템 데이터를 불러오는 중 오류 발생:', funSystemData.message);
+      }
+    
+    }catch(error){
+      console.error('펀시스템 데이터를 불러오는 중 오류 발생:', error);
+    }
+  };
+
+  useEffect(()=>{
+    window.addEventListener("scroll",handleScroll);
+
+    fetchNoticeData();
+    fetchFunSystemData();
+
+    return() =>{
+      window.removeEventListener("scroll",handleScroll);
+    }
+  }, [category, index, search]);
+
+  const handleSearch = (searchText : string) =>{
+    setPage(0);
+    setSearch(searchText);
+  }
+  const handleCategoryButton = (category :string) =>{
+    setPage(0);
+    setCategory(category);
+  }
+
+  const handleScroll = () => {
+    const scrollHeight = document.documentElement.scrollHeight;
+    const scrollTop = document.documentElement.scrollTop;
+    const clientHeight = document.documentElement.clientHeight;
+    if (scrollTop + clientHeight >= scrollHeight - 200 && !fetching) {
+      index ? fetchFunSystemData():fetchNoticeData(); //TODO : 검색까지 반영되게 
+    }
+   };
 
   const selectMenuHandler = (n: number) => {
     setIndex(n);
+    setPage(0);
+    setCategory("ALL");
+    setSearch("");
   };
 
   return (
     <styles.TabContentBox>
       <styles.Rectangle242>
-        {data.map((element, idx) => (
+        {["공지사항", "펀시스템"].map((title, idx) => (
           <styles.TabFontBox
-            key={element.id}
+            key={idx}
             isSelected={idx === index}
             onClick={() => selectMenuHandler(idx)}
           >
-            {element.title}
+            {title}
           </styles.TabFontBox>
         ))}
       </styles.Rectangle242>
-      <SearchBar/>
-      <ButtonGroup/>
-      {data[index].contents}
-     
+      <SearchBar onSearch={handleSearch}/>
+      <ButtonGroup index = {index} category={handleCategoryButton}/>
+
+      {index === 0 ? (
+        <NoticeList items={noticeData}/>
+      ) : (
+        <NoticeList items={funSystemData} />
+      )}
```

Comment:
이 부분을 추가하는 방식으로 하신건가요??
제가 코드를 제대로 이해한건지 궁금합니당!

Reply:
맞습니다!! 공지사항/펀시스템 각각 데이터 저장하는 data 부분이 필요없을 것 같아서 제목 배열에 바로 매핑해서 진행했습니다

```suggestion
{["공지사항", "펀시스템"].map((title, idx) => (
          <styles.TabFontBox
            key={idx}
            isSelected={idx === index}
            onClick={() => selectMenuHandler(idx)}
          >
            {title}
          </styles.TabFontBox>
        ))}
      </styles.Rectangle242>
      <SearchBar onSearch={handleSearch}/>
      <ButtonGroup index = {index} category={handleCategoryButton}/>

      {index === 0 ? (
        <NoticeList items={noticeData}/>
      ) : (
        <NoticeList items={funSystemData} />
      )}
```